dali/server/daserver.cpp

```diff
@@ -136,6 +139,199 @@ void usage(void)
     printf("--daemon|-d <instanceName>\t: run daemon as instance\n");
 }
 
+/* NB: Ideally this belongs within common/environment,
+ * however, that would introduce a circular dependency.
+ */
+static bool populateWhiteListFromEnvironment(IWhiteListWriter &writer)
+{
+    Owned<IRemoteConnection> conn = querySDS().connect("/Environment", 0, 0, INFINITE);
+    assertex(conn);
+
+    // only ever expecting 1 DaliServerProcess and 1 WhiteList
+    const IPropertyTree *whiteListTree = conn->queryRoot()->queryPropTree("Software/DaliServerProcess[1]/WhiteList[1]");
+    if (whiteListTree)
+    {
+        if (!whiteListTree->getPropBool("@enabled", true)) // on by default
+            return false;
+        // Default for now is to allow clients that send no role (legacy) to connect if their IP is whitelisted.
+        writer.setAllowAnonRoles(whiteListTree->getPropBool("@allowAnonRoles", true));
+    }
+
+    std::unordered_map<std::string, std::string> machineMap;
+
+    auto populateMachineMap = [&machineMap](const IPropertyTree &environment)
+    {
+        Owned<IPropertyTreeIterator> machineIter = environment.getElements("Hardware/Computer");
+        ForEach(*machineIter)
+        {
+            const IPropertyTree &machine = machineIter->query();
+            const char *name = machine.queryProp("@name");
+            const char *host = machine.queryProp("@netAddress");
+            machineMap.insert({name, host});
+        }
+    };
+    auto resolveComputer = [&machineMap](const char *compName, const char *defaultValue, StringBuffer &result) -> const char *
+    {
+        const auto &it = machineMap.find(compName);
+        if (it == machineMap.end())
+            return defaultValue;
+        IpAddress ip(it->second.c_str());
+        if (ip.isNull())
+            return defaultValue;
+        return ip.getIpText(result);
+    };
+    auto addRoles = [&writer, &resolveComputer](const IPropertyTree &component, const std::vector<unsigned __int64> &roles)
```

Comment:
Haven't used a std::initializer_list yet, but the std::vector is by passed by reference, so.. there should be no temporary copy.. so shouldn't be any more efficient?

Reply:
In fact having read about std::initializer_list, I think it will be more expensive (than currently simply passing by reference).
A initializer_list will create a temporary array:
{quote}
The underlying array is a temporary array of type const T[N], in which each element is copy-initialized (except that narrowing conversions are invalid) from the corresponding element of the original initializer list.
{quote}

Reply:
ok.  I suspect the initialiser is better - because it won't allocate anything on the heap.  I suspect it will generate a constant array, so have no creation overhead.

It will be interesting to compare the generated code.

Reply:
BTW The code you have will already be creating an initialiser - because that's passed as the parameter to the vector constructor.